Documentation/contributor-guide/release.md

```diff
@@ -37,48 +37,62 @@ All releases version numbers follow the format of [semantic versioning 2.0.0](ht
 - Put `[GH XXXX]` at the head of change line to reference Pull Request that introduces the change. Moreover, add a link on it to jump to the Pull Request.
 - Find PRs with `release-note` label and explain them in `NEWS` file, as a straightforward summary of changes for end-users.
 
-## Build and push the release artifacts
+## Patch release criteria
 
-- Ensure `docker` is available.
+The etcd project aims to release a new patch version if any of the following conditions are met:
 
-Run release script in root directory:
+- Fixed one or more major CVEs (>=7.5).
+- Fixed one or more critical bugs.
+- Fixed three or more major bugs.
+- Fixed five or more minor bugs.
 
+## Release guide
+### Prerequisites
+There are some prerequisites, which should be done before the release process. These are one time operations,
+which don't need to be executed before releasing each version.
+1. Generate a GPG key and add it to your github account. Refer to the links on [settings](https://github.com/settings/keys).
+2. Ensure you have a Linux machine, on which the git, golang and docker have been installed.
+    - Ensure the golang version matches the version defined in `.go-version` file.
+    - Ensure non-privileged users can run docker commands, refer to the [Linux postinstall](https://docs.docker.com/engine/install/linux-postinstall/).
+    - Ensure there is at least 500MB free space on your Linux machine.
+3. Install gsutil, refer to [gsutil_install](https://cloud.google.com/storage/docs/gsutil_install).
+4. Authenticate the image registry, refer to [Authentication methods](https://cloud.google.com/container-registry/docs/advanced-authentication).
+   - `gcloud auth login`
+   - `gcloud auth configure-docker`
+
+### Release steps
+1. Raise an issue to public the release plan, e.g. [issues/17350](https://github.com/etcd-io/etcd/issues/17350).
+2. Verify you can pass the authentication to the image registries,
+   - `docker login gcr.io`
+   - `docker login quay.io`
+3. Clone the etcd repository and checkout the target branch,
+   - `git clone git@github.com:etcd-io/etcd.git`
+   - `git checkout release-3.X`
+4. Run the release script under the repository's root directory,
+   - `DRY_RUN=false ./scripts/release.sh ${VERSION}`
```

Comment:
Should we be suggesting `DRY_RUN=true` first to confirm users machine is working as expected?

Reply:
I tend to keep it as it's for now. 
- Our github workflow confirms this on each PR and Push. 
- Even dry-run also has some side effect, e.g. it will generate binaries and images, but not pushing them to bucket and registries.